tiledb/sm/array/array_directory.cc

```diff
@@ -685,11 +685,8 @@ bool ArrayDirectory::timestamps_overlap(
   } else {
     // When consolidated fragment has timestamps, true if there is even partial
     // overlap
-    auto partial_overlap =
-        ((fragment_timestamp_start >= timestamp_start_ &&
-          fragment_timestamp_start <= timestamp_end_) ||
-         (fragment_timestamp_end >= timestamp_start_ &&
-          fragment_timestamp_end <= timestamp_end_));
+    auto partial_overlap = (fragment_timestamp_start <= timestamp_end_) &&
+                           (timestamp_start_ <= fragment_timestamp_end);
```

Comment:
Question, was this broken or is this an optimization?

Reply:
The previous version was broken. It was missing the case where we have partial overlap because we opened the array later than the fragment start time and closed it before the fragment end time. In my mind that was full overlap, because the array open-close times were fully covered so I excluded it, but in reality it's not, since it's the fragment overlap that we care for and the fragment is only partially covered in that case.